api/api.go

```diff
@@ -2,18 +2,43 @@ package api
 
 import (
 	"context"
+	"encoding/hex"
 	"fmt"
 	"math/big"
 
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/eth/filters"
 	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/onflow/flow-evm-gateway/storage"
+	"github.com/onflow/flow-go/fvm/evm/emulator"
 )
 
-type BlockChainAPI struct{}
+const EthNamespace = "eth"
```

Comment:
The JSON-RPC specification describes various namespaces, such as:
- `eth`,
- `debug`,
- `web3`,
- `net` etc

The general idea behind it is that you register a `Service` object under a certain `Namespace`, e.g.
```go
[]rpc.API{
	{
		Namespace: "eth",
		Service:   NewEthereumAPI(apiBackend),
	}, {
		Namespace: "eth",
		Service:   NewBlockChainAPI(apiBackend),
	}, {
		Namespace: "eth",
		Service:   NewTransactionAPI(apiBackend, nonceLock),
	}, {
		Namespace: "txpool",
		Service:   NewTxPoolAPI(apiBackend),
	}, {
		Namespace: "debug",
		Service:   NewDebugAPI(apiBackend),
	}, {
		Namespace: "eth",
		Service:   NewEthereumAccountAPI(apiBackend.AccountManager()),
	}, {
		Namespace: "personal",
		Service:   NewPersonalAccountAPI(apiBackend, nonceLock),
	},
}
```

When the JSON-RPC server receives a request such as:

```json
{"jsonrpc":"2.0","id":2,"method":"eth_blockNumber","params": []}
```
It will search through the `eth` namespace for a registered `Service` object that exposes a `BlockNumber()` method and call it.

I have added it as a constant, to avoid hard-coding it.